src/components/PopupWithForm.js

```diff
@@ -20,10 +20,12 @@ export default class PopupWithForm extends Popup {
       e.preventDefault();
       const originalText = this.#submitButton.textContent;
       this.#submitButton.textContent = "Saving...";
-      this._handleFormSubmit(this._getInputValues()).then(() => {
-        this.#submitButton.textContent = originalText;
-        this.close();
-      });
+      this._handleFormSubmit(this._getInputValues())
+        .then(() => {
+          this.close();
+        })
+        .catch(console.error)
+        .finally((this.#submitButton.textContent = originalText));
```

Comment:
`        .finally((this.#submitButton.textContent = originalText));
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well.